a11ym

```diff
@@ -243,46 +283,69 @@ crawler
     )
     .on(
         'fetcherror',
-        function () {
-            process.stderr.write(queueItem.url + ' failed to fetch.\n');
+        function (queueItem) {
+            process.stderr.write(
+                queueItem.url + ' failed to fetch ' +
+                '(status code: ' + queueItem.stateData.code + ')' +
+                '.\n'
+            );
         }
     ).
     on(
         'fetchcomplete',
         function(queueItem) {
-            console.log('Found: ' + queueItem.url + '.');
+            var url          = queueItem.url;
+            var parsedUrl    = parseURL(url);
+            var urlQueueName = (parsedUrl.pathname.split('/', 2)[1] || '__root__');
+
+            var logPrefix = '[' + urlQueueName + '] Fetched: ' + url;
 
             if (!queueItem.stateData.contentType ||
                 null === queueItem.stateData.contentType.match(/^text\/html/)) {
-                console.log('    (skipped, not text/html).');
+                console.log(logPrefix + '; skipped, not text/html.');
 
                 return;
             }
 
-            var url = queueItem.url;
-
             if (null !== filterByUrls && false === filterByUrls.test(url)) {
-                console.log('    (filtered).');
+                console.log(logPrefix + '; filtered.');
 
                 return;
             }
 
             if (null !== excludeByUrls && true === excludeByUrls.test(url)) {
-                console.log('    (excluded).');
+                console.log(logPrefix + '; excluded.');
 
                 return;
             }
 
-            if (--maximumUrls < 0) {
+            if (maximumUrls < 0) {
                 crawler.stop();
-                console.log(
-                    '    (ignored, maximum URLs reached).'
-                );
+                console.log(logPrefix + '; ignored, maximum URLs reached.');
 
                 return;
             }
 
-            urls.push({url: url});
+            console.log(logPrefix + '.');
+
+            if (undefined === urlQueues[urlQueueName]) {
+                urlQueues[urlQueueName] = async.queue(
+                    function (task, onTaskComplete) {
+                        console.log(chalk.magenta('[' + task.queueName + '] Waiting to run: ' + task.url) + '.');
+                        testQueue.push({
+                            url          : task.url,
+                            queueName    : task.queueName,
+                            onUrlComplete: onTaskComplete
```

Comment:
Maybe add a comment on how this works.

It's not immediately clear that `onTaskComplete` gets only called once the URL was processed by the testQueue.